core/src/avm1/globals/object.rs

```diff
@@ -123,6 +125,30 @@ fn value_of<'gc>(
     Ok(ReturnValue::Immediate(this.into()))
 }
 
+/// Implements `Object.registerClass`
+pub fn register_class<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    if let Some(class_name) = args
+        .get(0)
+        .and_then(|v| v.clone().coerce_to_string(avm, context).ok())
```

Comment:
`args.get(0).cloned().unwrap_or(Value::Undefined).coerce_to_string(...)?` because this could theoretically be called on a Object with a toString that does a `throw`, in which case it should bail out with ?.